Source/WebCore/platform/mediastream/mac/AVVideoCaptureSource.mm

```diff
@@ -617,27 +629,62 @@ static inline IntDegrees sensorOrientation(AVCaptureVideoOrientation videoOrient
 }
 ALLOW_DEPRECATED_DECLARATIONS_END
 
+bool AVVideoCaptureSource::lockForConfiguration()
```

Comment:
Should we also use it inside AVVideoCaptureSource::setSessionSizeFrameRateAndZoom()?

Reply:
I didn't initially because that is a bigger change, since we don't bail if we aren't able to lock, but I think configuration will fail in that case anyway so I might as well fix it now.